### ISSUE TITLE
Find alternative to destructive title→filename mapping

### ISSUE THREAD

**Issue by birjolaxew** [simple-icons/simple-icons#882]
This issue popped up in relation to #875, and is explained [here](https://github.com/simple-icons/simple-icons/pull/875#issuecomment-405777365).

The TL;DR is that our mapping removes some symbols (e.g. dots, hyphens and exclamation marks) when translating a title into a filename. This leads to some issues, mainly that .NET's filename becomes `net.svg` which isn't super clear, and potentially (although very unlikely) filename collisions in the future (if a brand was to be named e.g. `-NET`).

Possible solutions as I see it:

1. Leave it as-is. Although not ideal, it could be argued that `net.svg` is a reasonable representation of `.NET` especially since the filename doesn't have huge requirements for clarity - only for uniqueness. This would leave up the potential collision issue.

2. [Change mapping to not lose information](https://github.com/simple-icons/simple-icons/pull/875#issuecomment-405777365) - this would involve changing e.g. `"." → ""` to `"." → "dot"`. This would be a breaking change as some filenames would have to be changed (e.g. `aboutme.svg` becomes `aboutdotme.svg`) breaking CDN usage of those icons - the JS API would be unaffected. This would leave up, but greatly diminsh, the potential collision issue.

3. Implement explicit mapping somehow (e.g. as key in `_data`). This wouldn't be a breaking change as no external API would change, but would be a bit ugly code-wise (some icons having extra metadata while other's don't, and the metadata being a bit out of place in `_data`). It also seems to me like making a mountain out of a molehill.

I'm personally fondest of option 2 here. Any opinions @ericcornelissen / @davidklebanoff ?

---

*Blocks #875*

**Comment by ericcornelissen** [comment]
I'm okay with option 2 as well. I think option 3 is valid as well, but in that case I think we should just require this extra key for all entries in `_data`...
